src/lib/externalAttachment/index.ts

```diff
@@ -0,0 +1,92 @@
+import { singleton } from 'tsyringe'
+import { Logger } from 'pino'
+import { logger } from '../logger.js'
+import Identity from '../identity.js'
+import env from '../../env.js'
+import { AttachmentRow } from '../db/types.js'
+
+@singleton()
+export class ExternalAttachmentService {
+  private log: Logger
+
+  constructor(private identity: Identity) {
+    this.log = logger.child({ service: 'ExternalAttachment' })
+  }
+
+  async getOidcConfig(oidcConfigUrl: string) {
+    const response = await fetch(oidcConfigUrl)
+    if (!response.ok) {
+      throw new Error('Failed to fetch OIDC configuration')
+    }
+    return response.json()
```

Comment:
this looks to have an implicit `any` type as a return? We should validate the repsonse to make sure it contains the properties we want with `zod`